neqo-http3/src/recv_message.rs

```diff
@@ -94,11 +100,16 @@ impl RecvMessage {
         conn_events: Box<dyn RecvMessageEvents>,
         push_handler: Option<Rc<RefCell<PushController>>>,
         priority_handler: PriorityHandler,
+        header_frame_type_read: bool, // TODO Use stream PEEK for decoding a stream type instread of consuming data.
     ) -> Self {
+        let mut frame_reader = HFrameReader::new();
+        if header_frame_type_read {
+            let mut d = Encoder::default();
+            d.encode_varint(H3_FRAME_TYPE_HEADERS);
+            frame_reader.consume(Decoder::from(&d)).unwrap();
```

Comment:
This seems like it is wasteful.  We should just tell HFrameReader to skip straight to the bit where it reads the length.